src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,86 @@
+package uk.ac.cam.cl.dtg.isaac.api;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
+import jakarta.ws.rs.core.Response;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import uk.ac.cam.cl.dtg.isaac.dto.users.RegisteredUserDTO;
+import uk.ac.cam.cl.dtg.segue.api.UsersFacade;
+import uk.ac.cam.cl.dtg.segue.api.monitors.PasswordResetByEmailMisuseHandler;
+import uk.ac.cam.cl.dtg.segue.api.monitors.PasswordResetByIPMisuseHandler;
+
+import static org.easymock.EasyMock.*;
+import static org.junit.Assert.assertEquals;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.ANONYMOUS_USER;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.NUMBER_SECONDS_IN_MINUTE;
+
+public class UsersFacadeIT extends IsaacIntegrationTest {
+    private UsersFacade usersFacade;
+
+    @BeforeClass
+    public static void beforeAll() {
+        misuseMonitor.registerHandler(PasswordResetByEmailMisuseHandler.class.getSimpleName(),
+                new PasswordResetByEmailMisuseHandler(1, 2, NUMBER_SECONDS_IN_MINUTE));
+        misuseMonitor.registerHandler(PasswordResetByIPMisuseHandler.class.getSimpleName(),
+                new PasswordResetByIPMisuseHandler(emailManager, properties, 1, 2, NUMBER_SECONDS_IN_MINUTE));
+    }
+
+    @Before
+    public void beforeEach() throws Exception {
+        misuseMonitor.resetMisuseCount("test-student@test.com", PasswordResetByEmailMisuseHandler.class.getSimpleName());
+        misuseMonitor.resetMisuseCount("0.0.0.0", PasswordResetByIPMisuseHandler.class.getSimpleName());
+        this.usersFacade = new UsersFacade(properties, userAccountManager, logManager, userAssociationManager, misuseMonitor, userPreferenceManager, schoolListReader);
+    }
+
+    @Test
+    public void resetPassword_emailRateLimits() {
+        HttpSession mockSession = createNiceMock(HttpSession.class);
```

Comment:
Same problem as in the other file. In fact they're both very similar. Is it possible to reduce all this repetition somehow?